container/new_pull.bzl

```diff
@@ -0,0 +1,208 @@
+              " is not defined, the home directory will be used.",
+        mandatory = False,
+    ),
+    "os": attr.string(
+        default = "linux",
+        doc = "(optional) Which os to pull if this image refers to a " +
+              "multi-platform manifest list, default 'linux'.",
+    ),
+    "os_features": attr.string_list(
+        doc = "(optional) Specifies os features when pulling a multi-platform " +
+              "manifest list.",
+    ),
+    "os_version": attr.string(
+        doc = "(optional) Which os version to pull if this image refers to a " +
+              "multi-platform manifest list.",
+    ),
+    "platform_features": attr.string_list(
+        doc = "(optional) Specifies platform features when pulling a " +
+              "multi-platform manifest list.",
+    ),
+    "registry": attr.string(
+        mandatory = True,
+        doc = "The registry from which we are pulling.",
+    ),
+    "repository": attr.string(
+        mandatory = True,
+        doc = "The name of the image.",
+    ),
+    "tag": attr.string(
+        default = "latest",
+        doc = "(optional: currently unsupported) The tag of the image, default to 'latest' " +
+              "if this and 'digest' remain unspecified.",
+    ),
+    "_puller": attr.label(
+        executable = True,
+        default = Label("@go_puller//file:downloaded"),
+        cfg = "host",
+    ),
+}
+
+def _impl(repository_ctx):
+    """Core implementation of container_pull."""
+
+    # Add an empty top-level BUILD file.
+    repository_ctx.file("BUILD", "")
+
+    # TODO(xwinxu): change container_import rule to have an oci attribute to account for new OCI Image Layout
+    # currently exclude:     exports_files(["image.digest", "digest"]) in BUILD, theoretical BUILD commented out below:
+    repository_ctx.file("image/BUILD", """
+package(default_visibility = ["//visibility:public"])
+
+load("@io_bazel_rules_docker//container:import.bzl", "container_import")
+
+container_import(
+    name = "image",
+    config = "image/index.json",
+    layers = glob(["image/blobs/sha256/*"]),
+)
+    """)
+
+    args = [
+        repository_ctx.path(repository_ctx.attr._puller),
+        "-directory",
+        repository_ctx.path("image"),
+        "-os",
+        repository_ctx.attr.os,
+        "-os-version",
+        repository_ctx.attr.os_version,
+        "-os-features",
+        " ".join(repository_ctx.attr.os_features),
+        "-architecture",
+        repository_ctx.attr.architecture,
+        "-variant",
+        repository_ctx.attr.cpu_variant,
+        "-features",
+        " ".join(repository_ctx.attr.platform_features),
+    ]
+
+    # Use the custom docker client config directory if specified.
+    if repository_ctx.attr.docker_client_config != "":
+        args += ["-client-config-dir", "{}".format(repository_ctx.attr.docker_client_config)]
+
+    # TODO(xwinxu): There is currently no support for caches in rules_docker. Stay tuned...
```

Comment:
I would probably remove this part first. We still need to figure out the cache story for go-containerregistry.